rollout/trafficrouting.go

```diff
@@ -163,9 +167,13 @@ func (c *rolloutContext) reconcileTrafficRouting() error {
 			} else if c.rollout.Status.Canary.Weights != nil {
 				desiredWeight = c.rollout.Status.Canary.Weights.Canary.Weight
 			}
+
+			err := reconciler.RemoveManagedRoutes()
+			if err != nil {
+				return err
+			}
 		} else if index != nil {
 			atDesiredReplicaCount := replicasetutil.AtDesiredReplicaCountsForCanary(c.rollout, c.newRS, c.stableRS, c.otherRSs, nil)
-			setHeaderRouting = replicasetutil.GetCurrentSetHeaderRouting(c.rollout, *index)
```

Comment:
This will find the last set of setHeaderRoute and modify it even if we are on the next step.
For the case when we're on step 2 and VS was modified the system can recover the state of VS.
```yaml
steps:
- setHeaderRoute: "Agent=Chome"
- pause {} # step 2
```

The question do we want to have that "recovering" functionality for this. At least we have it for the `setWeight` step.

If we don't need that then `GetCurrentSetHeaderRouting` should be removed as it's not using anymore.

Reply:
@zachaller 
the `GetCurrentSetHeaderRouting` method can be removed as it is not used anymore, just in tests.